server/play-server/src/main/scala/tapir/server/play/PlayDecodeInputContext.scala

```diff
@@ -0,0 +1,32 @@
+package tapir.server.play
+
+import akka.stream.Materializer
+import play.api.mvc.RequestHeader
+import tapir.internal.server.DecodeInputsContext
+import tapir.model.{Method, ServerRequest}
+
+private[play] class PlayDecodeInputContext(request: RequestHeader, pathConsumed: Int = 0, serverOptions: PlayServerOptions)(
+    implicit mat: Materializer
+) extends DecodeInputsContext {
+  override def method: Method = Method(request.method.toUpperCase())
+
+  override def nextPathSegment: (Option[String], DecodeInputsContext) = {
+    val path = request.path.drop(pathConsumed)
+    val nextStart = path.dropWhile(_ == '/')
+    val segment = nextStart.split("/", 2) match {
+      case Array("")   => None
+      case Array(s)    => Some(s)
+      case Array(s, _) => Some(s)
+    }
+    val charactersConsumed = segment.map(_.length).getOrElse(0) + (path.length - nextStart.length)
+
+    (segment, new PlayDecodeInputContext(request, pathConsumed + charactersConsumed, serverOptions))
+  }
+  override def header(name: String): List[String] = request.headers.toMap.get(name).toList.flatten
+  override def headers: Seq[(String, String)] = request.headers.headers
+  override def queryParameter(name: String): Seq[String] = request.queryString.get(name).toSeq.flatten
+  override def queryParameters: Map[String, Seq[String]] = request.queryString
+  override def bodyStream: Any = ???
```

Comment:
How to get a body stream here? I found that there is even an `Accumulator.source` but every accumulator evaluates to a `Future`. How do we get a `Source[Byte, _]` from it?

Reply:
Pretty sure that Play framework does not support input body streaming. How should we handle this case, what do we want to return here?

Reply:
Thanks for a clarification. As discussed with @adamw we will throw an exception here.